### ISSUE TITLE
Use "postgres" as Neon default database name and the associated user "postgres", too

### ISSUE THREAD

**Issue by Bodobolero** [neondatabase/neon#9452]
Upstream PostgreSQL initdb creates a default database postgres and user postgres.

According to https://www.postgresql.org/docs/17/app-initdb.html 

_The postgres database is a default database meant for use by users, utilities and third party applications._

New customers coming from other providers and their own deployments of upstream postgres would have less surprises if Neon sticks to that convention.

Currently Neon creates a default database "neondb" with user "neondb_owner".

In Neon the "postgres" database exists but is used for Neon internal purposes and cannot be used by end users.
This is confusing and violates the principle of "least surprise".

Currently, we install some extensions and a healthcheck table and stuff like that in the postgres database.

The following options exist
- put Neon internal objects (extensions, healtcheck table etc.) in pg_catalog schema or similar schema
- create a separate database with a name that implies "for Neon usage only" that is used for management, e.g. "neon_service"

Optional:
- Tristan suggested to not allow users to create objects with prefix `neon_`

Benefits:
- make it easier for new users who are already familiar with postgres
- shorter connection strings (user postgres and database postgres are defaults)

Some implementation considerations
- we need to decide what to do with existing projects that use neondb and neondb_owner (do not break existing customer applications, nor their connection strings)

Notes:
there has been discussion in two Neon internal slack channels on this topic
[#team-compute](https://neondb.slack.com/archives/C04DGM6SMTM/p1729253412157579)
[#postgres ](https://neondb.slack.com/archives/C03QLRH7PPD/p1729252247405989)

**Comment by ololobus** [comment]
> Upstream PostgreSQL initdb creates ... and user postgres.

That's just not true, as far as I remember. `initdb` takes the OS user by default unless it's specified as an arg. The default role is also a superuser, so it doesn't matter much for the end user. Either way, the name of the default role we give to user is configured from the control plane side, so there is nothing compute team should do here (if you want to change the name of the first user role to `postgres`, which I still do not know why)

Regarding the default DB I will leave here the same comment I've left in the Slack

> We allow specifying the default DB via API and iirc some partners use this feature, maybe Hasura. DB `postgres` is hardcoded in the `initdb`, so if one would specify `mydb`, then we would create 3 DBs: `mydb`, `postgres` (which you never asked for) and `neon_service`. That looks awkward to me
> 
> Another reason is backward compatibility. We already have a ton of DBs with `postgres` used as a service DB. If we change it to `neon_service` or anything, there will be two options
> 1. Continue using `postgres` for old projects and switch to `neon_service` for the new ones
> 2. Migrate old projects to `neon_service` lazily too, this will be doomed if at least one user managed to create `neon_service` already
> 
> Anyway, in both cases, that will be a messy code in several places with a bunch of ifs, which is bug-prone
> 
> That said, we have a moderate amount of work with some drawbacks without much of a clear benefit

**Comment by Bodobolero** [comment]
```bash
New EC machine (Debian 12)

ssh admin@18.199.103.251
# now I am user OS "admin"
sudo apt update
sudo apt install postgresql # creates user postgres, runs initdb as user postgres
sudo -i -u postgres
psql
psql (15.8 (Debian 15.8-0+deb12u1))
Type "help" for help.
postgres=# \l
                                             List of databases
   Name    |  Owner   | Encoding | Collate |  Ctype  | ICU Locale | Locale Provider |   Access privileges   
-----------+----------+----------+---------+---------+------------+-----------------+-----------------------
 postgres  | postgres | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | 
 template0 | postgres | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =c/postgres          +
           |          |          |         |         |            |                 | postgres=CTc/postgres
 template1 | postgres | UTF8     | C.UTF-8 | C.UTF-8 |            | libc            | =c/postgres          +
           |          |          |         |         |            |                 | postgres=CTc/postgres
(3 rows)
```
==> database postgres
==> OS user postgres (created by installation)
==> owner postgres

**Comment by ololobus** [comment]
> sudo apt install postgresql # creates user postgres, runs initdb as user postgres

Yes, that's a package / distribution feature, not something that initdb does, so it doesn't contradict what I replied, but I got your point.
